src/scripts/modules/add_destination.js

```diff
@@ -0,0 +1,130 @@
+import destinationsList from "../destinationList.js";
+import events from "./pubsub.js";
+
+let form;
+let uploadBtn;
+let radioBtns;
+let radioVisited;
+let radioBucketlist;
+let cityInput;
+let countryInput;
+let photoInput;
+let newCity;
+let newCountry;
+let newPhoto;
+// let updatedList
+
+function cacheDom() {
+  form = document.getElementById("add-city-form");
+  uploadBtn = document.getElementById("upload");
+  radioBtns = document.querySelectorAll("input[type=radio]");
+  radioVisited = document.getElementById("rd-visited");
+  radioBucketlist = document.getElementById("rd-bucketlist");
+  cityInput = document.getElementById("inputCity");
+  countryInput = document.getElementById("inputCountry");
+  photoInput = document.getElementById("inputImg");
+}
+
+function bindEvents() {
+  form.addEventListener("submit", (event) => {
+    setValues();
+    addDestination();
+    event.preventDefault();
+    console.log(destinationsList);
+  });
+
+  // events.publish('destinationAdded', (form.addEventListener("submit", (event) => {
```

Comment:
What made you think you added to pass the form event listener to publish?

Reply:
As I mentioned I got stuck trying to get the publish data. My thinking was that the list creation happens when functions are called on submit so I tried a bunch of different ways to access or store that value that I now know don't work. Another tutorial used named modules and that seemed pretty straightforward so I was tempted to try that. That's what I see as the problem, where do you think I missed the point?

Reply:
> I got stuck trying to get the publish data

here's the trick. you don't need to publish the data, you need to publish the event on the right moment. in this case, publish the event after submitting (aka adding to the list of destinations) the new item. we almost don't even need to pass any data because the logic is simple and both modules have access to the full `destinationsList`. Still, I suggested passing the index (as I said in the other comment) just to make things easier.

This pubsub pattern can be tricky to "master", it's totally okay. Once again, remember, with practice you'll not only learn it but use it in a intuitive way :)

Reply:
Yeah it will click at some point :) I'll just practice it on something smaller, I should have done that in the first place

Reply:
Yes, practicing in something smaller before "the big boss" is always a good idea ;)